File: ExpenseApi/ExpenseApi/Models/Transaction.cs
using System;

namespace ExpenseApi.Models
{
    public class Transaction
    {
        public int Id { get; set; }
        public string Description { get; set; } = "";
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string Category { get; set; } = "";
        public TransactionType Type { get; set; }
    }

    public enum TransactionType
    {
        Income,
        Expense
    }
}

File: ExpenseVue/src/types/transaction.ts
/**
 * Represents a transaction in the system
 */
export interface Transaction {
  id: number;
  amount: number;
  description: string;
  category: string;
  date: string;
  type: TransactionType;
}

export enum TransactionType {
  Income = 0,
  Expense = 1,
}

/**
 * Filter type for transactions
 */
export type FilterType = "all" | "day" | "month" | "year";

/**
 * Formatted transaction with proper types
 */
export interface FormattedTransaction extends Transaction {
  amount: number;
  type: TransactionType;
  date: string;
}
